### PR TITLE
switched to pbdZMQ from rzmq

### LINKED ISSUE #233: reimplement with different zmq lib?
According to https://github.com/armstrtw/rzmq/pull/16#issuecomment-165628017, there is now a new zmq lib available on R on Linux, mac and windows and it's on CRAN: [pbdZMQ](https://cran.r-project.org/web/packages/pbdZMQ/index.html). If we implement the zmq stuff on this lib, we can upload irkernel and the rest to cran and get the windows and mac binary packages auto-build and we don't need to maintain our own repo anymore.

### PR DESCRIPTION
**Pull request by flying-sheep**
pls test rigorously:

i have no idea what i’m doing

i basically just changed stuff until the errors went away, since i have no idea how the jupyter protocol works and what the difference between “just sending data” and “messages” is in ZMQ.

---

if/when it really works, this should help with our rzmq linking and distributing woes.

fixes #233 and many more

@takluyver can you please upgrade my status in this organization so i can do releases and change descriptions and stuff?

PS: you need git master of pbdZMQ (which has polling support)

### CODE REVIEW

**Comment by flying-sheep** [comment]
pbdZMQ is not available with conda. should we use “install.packages” for it?

**Comment by JanSchulz** [comment]
I suspect that the conda people will be happy to replace the package (the windows package builds from a year old unmerged pr), so I wouldn't worry about it...

**Comment by flying-sheep** [comment]
what PR?

**Comment by rgbkrk** [comment]
@flying-sheep https://github.com/armstrtw/rzmq/pull/16

**Comment by flying-sheep** [comment]
well, since pbdZMQ is actively maintained and has up-to-date binary packages, i guess going with that one is better in the long run.

**Comment by rgbkrk** [comment]
Sure seems like it!

**Comment by flying-sheep** [comment]
so, does this work for everyone?

**Comment by rgbkrk** [comment]
I'm a fan, though I really don't have any say here. ;)

File: DESCRIPTION [hunk 1]
```diff
@@ -21,7 +21,7 @@ Imports:
     methods,
     evaluate (>= 0.5.4),
     IRdisplay,
-    rzmq (>= 0.7.0),
+    pbdZMQ (>= 0.2.1),
```
Comment (takluyver) [inline DESCRIPTION]: Just to check, this is a requirement on a version that's not yet released on CRAN, right? I'd like to wait for a release and then update the Travis instructions to build it before we merge this.

Comment (flying-sheep) [inline DESCRIPTION]: yep, not on CRAN. no matter if it'll exist or the next version will be 0.3.0, this will work :smiley:

Comment (rgbkrk) [inline DESCRIPTION]: If it's not on CRAN, I truly hope the CRAN gods let it get through easily...

Comment (flying-sheep) [inline DESCRIPTION]: well, i think @snoweye simply hasn’t released it yet

**Pull request merged**